### PR TITLE
Add Timed based stage

### PR DESCRIPTION
**Pull request by naferx**
References #2  
https://github.com/akka/akka/pull/20330
https://github.com/akka/akka/issues/19834

### CODE REVIEW

**Comment by patriknw** [comment]
would it be possible to include the tests also?

**Comment by naferx** [comment]
Yeah @patriknw, I'll try

**Comment by patriknw** [comment]
thanks

**Comment by patriknw** [comment]
@naferx any update on the tests?

**Comment by naferx** [comment]
sorry. I am so busy and looking for some free time to do it.

**Comment by naferx** [comment]
Hi @patriknw, I updated this PR and I started to implement the tests based on stream base class, but I don't know how to assert the results. Can you guide me ?

File: src/test/scala/akka/stream/contrib/TimedSpec.scala [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+      source.sendNext(3)
+      sink.expectNext(1, 3, 6)
+      source.sendComplete()
+      sink.expectComplete()*/
+      /*
+
+      val n = 20
+      val testRuns = 1 to 2
+
+      def script = Script((1 to n) map { x ⇒ Seq(x) → Seq(x) }: _*)
```
Comment (patriknw) [inline src/test/scala/akka/stream/contrib/TimedSpec.scala]: ah, it used the scipted tests. We should use a more simple approach here.

File: src/test/scala/akka/stream/contrib/TimedSpec.scala [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    import akka.stream.contrib.Implicits.TimedFlowDsl
+
+    "measure time it between elements matching a predicate" in {
+
+      val measureBetweenEvery = 5
+      val printInfo = (interval: Duration) ⇒ {
+        info(s"Measured interval between $measureBetweenEvery elements was: $interval")
+      }
+
+      val flow = Flow[Int].map(identity).timedIntervalBetween(_ % measureBetweenEvery == 0, printInfo)
```
Comment (patriknw) [inline src/test/scala/akka/stream/contrib/TimedSpec.scala]: can't you use `val testActor = TestProbe()` and send to that from `printInfo`?
it doesn't have to verify the actual timing, just that things are emitted

**Comment by naferx** [comment]
Hi @patriknw, sorry for the delay.
I've finally updated the tests. I hope I didn't forget something.

**Comment by 2m** [comment]
LGTM. I am going to move files to their new location (after #39), and then merge.

**Comment by 2m** [comment]
Superseded by #44

**Pull request closed without merging**